File: QE-Selenium-Java/Activity7.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity7 {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        // Open page
        driver.get("https://training-support.net/webelements/dynamic-controls");

        // Print title
        System.out.println("Title: " + driver.getTitle());

        // Check input field before enabling
        System.out.println(driver.findElement(By.xpath("//*[@id='textInput']")).isEnabled());
        driver.findElement(By.xpath("//*[@id='textInputButton']")).click();
        wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//*[@id='textInput']")));
        System.out.println(driver.findElement(By.xpath("//*[@id='textInput']")).isEnabled());
        driver.quit();
    }
}

File: QE-Selenium-Python/Activity7.py
from selenium import webdriver
from selenium.webdriver.common.by import By
with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/dynamic-controls")
    print(driver.title)
    print(driver.find_element(By.XPATH,"//input[@id='textInput']").is_enabled())
    driver.find_element(By.XPATH,"//button[contains(@id,'textInputButton')]").click()
    print(driver.find_element(By.XPATH,"//input[@id='textInput']").is_enabled())
    driver.quit()